akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -883,6 +1015,22 @@ class ClusterSingletonManager(
       if (m.uniqueAddress == cluster.selfUniqueAddress)
         logInfo("Self downed, waiting for removal")
       stay
+    case Event(ReleaseLeaseResult(result), _) ⇒
+      // TODO we could retry if false
+      logInfo("Lease release result [{}] in state [{}]", result, stateName)
+      stay
+    case Event(ReleaseLeaseFailure(t), _) ⇒
+      // TODO we could retry
+      log.error(t, "Failed to release lease. Singleton may not be able to run on another node until lease timeout occurs")
```

Comment:
might be good to retry, but on the other hand releasing should only be done when the ActorSystem is about to shutdown anyway so we don't have much time for retries. Possibly useful for the graceful leaving case where we might have a few seconds.

Reply:
how do we handle this in the lease implementation? do we stop heartbeating when there has been a release attempt?

Reply:
Yes we cancel the timer for heart beat when leaving the granted state, i'll leave this as a possible TODO for now